crates/relayer/src/channel.rs

```diff
@@ -1469,6 +1583,703 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Channel<ChainA, ChainB> {
         }
     }
 
+    pub fn build_chan_upgrade_try(&self) -> Result<Vec<Any>, ChannelError> {
+        let src_channel_id = self
+            .src_channel_id()
+            .ok_or_else(ChannelError::missing_local_channel_id)?;
+        let src_port_id = self.src_port_id();
+        let src_latest_height = self
+            .src_chain()
+            .query_latest_height()
+            .map_err(|e| ChannelError::chain_query(self.src_chain().id(), e))?;
+
+        let dst_channel_id = self
+            .dst_channel_id()
+            .ok_or_else(ChannelError::missing_local_channel_id)?;
+        let dst_port_id = self.dst_port_id();
+
+        // Fetch the src channel end that will be upgraded by the upgrade handshake
+        // Querying for the Channel End now includes the upgrade sequence number
+        let (channel_end, _) = self
+            .src_chain()
+            .query_channel(
+                QueryChannelRequest {
+                    port_id: src_port_id.clone(),
+                    channel_id: src_channel_id.clone(),
+                    height: QueryHeight::Specific(src_latest_height),
+                },
+                IncludeProof::Yes,
+            )
+            .map_err(|e| ChannelError::query(self.src_chain().id(), e))?;
+
+        // Building the channel proof at the queried height
+        let src_proof = self
+            .src_chain()
+            .build_channel_proofs(
+                &src_port_id.clone(),
+                &src_channel_id.clone(),
+                src_latest_height,
+            )
+            .map_err(ChannelError::channel_proof)?;
+
+        let (dst_channel_end, _) = self
+            .dst_chain()
+            .query_channel(
+                QueryChannelRequest {
+                    port_id: dst_port_id.clone(),
+                    channel_id: dst_channel_id.clone(),
+                    height: QueryHeight::Latest,
+                },
+                IncludeProof::Yes,
+            )
+            .map_err(|e| ChannelError::query(self.src_chain().id(), e))?;
+
+        let (upgrade, maybe_upgrade_proof) = self
+            .src_chain()
+            .query_upgrade(
+                QueryUpgradeRequest {
+                    port_id: self.src_port_id().to_string(),
+                    channel_id: src_channel_id.to_string(),
+                },
+                src_latest_height,
+            )
+            .map_err(|e| ChannelError::chain_query(self.src_chain().id(), e))?;
+
+        let upgrade_proof = maybe_upgrade_proof.ok_or(ChannelError::missing_upgrade_proof())?;
+
+        let proof_upgrade =
+            CommitmentProofBytes::try_from(upgrade_proof).map_err(ChannelError::malformed_proof)?;
+
+        // TODO: Is this check necessary?
```

Comment:
Since we retrieve the `channel_end` from a query using the `Channel`'s `src_port_id` and `src_channel_id`, if this assertion fails it would indicate an issue with:
* `Channel`'s `src_channel_id()` method
* `Channel`'s `src_port_id()` method
* `Channel`'s `dst_port_id()` method
* `query_channel()`
So I don't think the assertion is required here

Reply:
Ok great, let's remove this check then